src/csharp/BUILD-INTEGRATION.md

```diff
@@ -1,351 +1,93 @@
+<PackageReference Include="Grpc.Tools" Version="2.50.0">
+  <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
+  <PrivateAssets>all</PrivateAssets>
+</PackageReference>
 ```
 
-#### My .proto files have same filename in different folders
-
-Starting from Grpc.Tools version 2.31, protocol buffers compilation preserves original folder structure for generated files. Eg.
-
-- `../ProjectFolder/Protos/v2/http.proto`
-- `../ProjectFolder/Protos/v3/http.proto`
-
-Will result in:
+Support is provided for the following platforms:
+* Windows (x86, x64, and arm64 via using the x86 binaries)
+* MacOS (arm64)
+* Linux (x86, x64, and arm64)
 
-- `../ProjectFolder/obj/CONFIGURATION/FRAMEWORK/Protos/v2/Greet.cs`
-- `../ProjectFolder/obj/CONFIGURATION/FRAMEWORK/Protos/v2/GreetGrpc.cs`
-- `../ProjectFolder/obj/CONFIGURATION/FRAMEWORK/Protos/v3/Greet.cs`
-- `../ProjectFolder/obj/CONFIGURATION/FRAMEWORK/Protos/v3/GreetGrpc.cs`
+## Adding `.proto` files to a project
 
-This feature resolves problems we have faced in large projects. Moreover, There is now also a project-wide new option Protobuf_ProtoRoot to define the fallback ProtoRoot. If the ProtoRoot is set, this also reduces the amount of problems that lead to duplicates. Eg.
+To add `.proto` files to a project edit the project’s `.csproj` file and add an item group with a `<Protobuf>` element that refers to the `.proto` file, e.g.
 
 ```xml
-  <ItemGroup>
-    <Protobuf Include="Protos\v2\greet.proto" ProtoRoot="Protos" />
-  </ItemGroup>
+<ItemGroup>
+    <Protobuf Include="Protos\greet.proto" />
+</ItemGroup>
 ```
 
-Before Grpc.Tools version 2.31 all .proto files were compiled into `obj` directory, flattening relative paths. For proto files with duplicated names it cause following errors `NETSDK1022 Duplicate 'Compile' items were included. [...]` or `MSB3105 [...] Duplicate items are not supported by the "Sources" parameter`. The workaround for this problem was introducing relative paths in your `obj` folder, by manipulating output path. Eg. 
+Wildcards can be used to select several `.proto` files, e.g.
 
 ```xml
-  <ItemGroup>
-    <Protobuf Include="Protos/v2/http.proto" OutputDir="$(Protobuf_OutputPath)%(RelativeDir)"  />
-    <Protobuf Include="Protos/v3/http.proto" OutputDir="$(Protobuf_OutputPath)%(RelativeDir)"  />
-  </ItemGroup>
-```
-
-__Note, this was a workaround approach, we recommend updating Grpc.Tools to the latest version.__
-
-### I just want to generate proto and gRPC C# sources from my .proto files (no C# compile)
-
-Suppose you want to place generated files right beside each respective source
-.proto file. Create a .csproj library file in the common root of your .proto
-tree, and add a reference to Grpc.Tools package (this works in Windows too, `$`
-below stands for a command prompt in either platform):
-
+<ItemGroup>
+    <Protobuf Include="**\*.proto" />
```

Comment:
This doc doesn't mention the option to enable auto-inclusion of .proto files anywhere.
See https://github.com/grpc/grpc/blob/master/src/csharp/Grpc.Tools/implementation_notes.md#automatically-including-proto-files  - perhaps worth mentioning this somewhere?